experimental/devices/inky/inky.go

```
@@ -0,0 +1,291 @@
+func (d *Dev) String() string {
+	return "InkyPHat"
+}
+
+// Halt implements conn.Resource
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// ColorModel implements display.Drawer
+// Maps white to white, black to black and anything else as red. Red is used as
+// a placeholder for the display's third color, i.e., red or yellow.
+func (d *Dev) ColorModel() color.Model {
+	return color.ModelFunc(func(c color.Color) color.Color {
+		r, g, b, _ := c.RGBA()
+		if r == 0 && g == 0 && b == 0 {
+			return color.RGBA{
+				R: 0,
+				G: 0,
+				B: 0,
+				A: 255,
+			}
+		} else if r == 0xffff && g == 0xffff && b == 0xffff {
+			return color.RGBA{
+				R: 255,
+				G: 255,
+				B: 255,
+				A: 255,
+			}
+		}
+		return color.RGBA{
+			R: 255,
+			G: 0,
+			B: 0,
+			A: 255,
+		}
+	})
+}
+
+// Bounds implements display.Drawer
+func (d *Dev) Bounds() image.Rectangle {
+	return image.Rect(0, 0, rows, cols)
+}
+
+// Draw implements display.Drawer
+func (d *Dev) Draw(dstRect image.Rectangle, src image.Image, srcPtrs image.Point) error {
+	if dstRect != d.Bounds() {
+		return fmt.Errorf("Partial update not supported")
+	}
+
+	if src.Bounds() != d.Bounds() {
+		return fmt.Errorf("Image must be the same size as bounds: %v", d.Bounds())
+	}
+
+	b := src.Bounds()
+	// Black/white pixels.
+	white := make([]bool, rows*cols)
+	// Red/Transparent pixels.
+	red := make([]bool, rows*cols)
+	for x := b.Min.X; x < b.Max.X; x++ {
+		for y := b.Min.Y; y < b.Max.Y; y++ {
+			i := x*cols + y
+			srcX := x
+			srcY := b.Max.Y - y - 1
+			r, g, b, _ := d.ColorModel().Convert(src.At(srcX, srcY)).RGBA()
+			if r == 0xffff && g == 0xffff && b == 0xffff {
+				white[i] = true
+			} else if r == 0xffff {
+				// Red pixels also need white behind them.
+				white[i] = true
+				red[i] = true
+			}
+		}
+	}
+
+	bufA, _ := pack(white)
+	bufB, _ := pack(red)
+	return d.update(borderColor[d.border], bufA, bufB)
+}
+
+func (d *Dev) update(border byte, black []byte, red []byte) error {
+	d.reset()
```

Please check for errors everywhere, would it be for consistency. And yes reset() should return an error.

I understand the code below will become a mess though, in theory some of it could be moved into a table + a loop, where you create a table and then iterate over the commands in it, i.e.

```
init := []struct{
  cmd byte
  data[]byte
}{
  {0x74, []byte{0x54}},
  {0x7e, []byte{0x3b}},
  ...
}
```

That said I don't want you to spend too much time on this, so it's not a big deal.